docs/saleor-webhook.md

```diff
@@ -0,0 +1,305 @@
+# Saleor Webhook utilities
+
+Apps are usually connected via webhooks - one App sends an HTTP request to another App, informing about some event or requesting some action to be performed.
+
+To inform your App about events originated from Saleor, you need to expose a webhook handler, which Saleor will call with POST request.
+
+To avoid boilerplate, App SDK provides utility that abstracts connection details, allowing developers to focus on business logic.
+
+There are 2 working classes
+
+- `SaleorAsyncWebhook`
+- `SaleorSyncWebhook`
+
+## Common configuration
+
+Both `SaleorSyncWebhook` and `SaleorAsyncWebhook` contain similar API with little differences.
+
+### Constructing Webhook instance
+
+In Next.js pages create a page, e.g. `pages/api/webhooks/order-created.ts`. We recommend to keep webhook type in file name, which will be resolved by Next.js to route path.
+
+```typescript
+import { SaleorAsyncWebhook } from "@saleor/app-sdk/handlers/next";
+
+/**
+ * To be type safe, define payload from API. This should be imported from generated graphQL code
+ */
+type OrderPayload = {
+  id: string;
+};
+
+export const orderCreatedWebhook = new SaleorAsyncWebhook<OrderPayload>(options);
+```
+
+For `SaleorSyncWebhook` it will be similar. Create e.g. `order-calculate-taxes.ts` page and create a new instance:
+
+```typescript
+import { SaleorSyncWebhook } from "@saleor/app-sdk/handlers/next";
+
+/**
+ * To be type safe, define payload from API. This should be imported from generated graphQL code
+ */
+type CalculateTaxedPayload = {
+  // taxes payload from subscription
+};
+
+export const orderCalculateTaxesWebhook = new SaleorSyncWebhook<OrderPayload>(options);
+```
+
+### Configuring Webhook instance
+
+`options` in constructor must be specified. Here are all options:
+
+```typescript
+type Options = {
+  /**
+   * Additional webhook name, optional.
+   */
+  name?: string;
+  /**
+   * Path to webhook. Should represent relative path from base app URL. In Next.js i will be usually starting with "api/".
```

Comment:
```suggestion
   * Path to webhook. Should represent relative path from base app URL. In Next.js it will start with `api/`, e.g. `api/webhooks/order-created`.
```

All backend endpoints routes start with `/api`